### PR TITLE
Review of the MetaPhlAn4

### PR DESCRIPTION
**Pull request by LilyAnderssonLee**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/taxprofiler/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/taxprofiler _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by LilyAnderssonLee** [comment]
@jfy133  Could you please change the `MetaPhlAn3 ` into `MetaPhlAn` on the workflow chart within the file `docs/output.md`?

File: docs/usage.md [hunk 1]
```diff
@@ -716,22 +716,22 @@ See the [MALT manual](https://software-ab.informatik.uni-tuebingen.de/download/m
 
 MetaPhlAn does not allow (easy) construction of custom databases. Therefore we recommend to use the prebuilt database of marker genes that is provided by the developers.
 
-To do this you need to have `MetaPhlAn` installed on your machine.
+To perform this task, ensure that you have installed `MetaPhlAn` on your machine. Keep in mind that each version of MetaPhlAn aligns with a specific version of the database. Therefore, if you download the MetaPhlAn3 database, remember to include `--mpa3` as a parameter for the database.
```
Comment (jfy133) [inline docs/usage.md:719]:
```suggestion
To perform this task, ensure that you have installed `MetaPhlAn` on your machine. Keep in mind that each version of MetaPhlAn aligns with a specific version of the database. Therefore, if you download the MetaPhlAn3 database, remember to include `--mpa3` as a parameter for the database in the `--databases` CSV file.
```

**Review by jfy133 — APPROVED** [review]
One tiny tweak and you're good to go :)

**Comment by jfy133** [comment]
Here are the files, you'll need to replace the ones in in `docs/images` with the following:

taxprofiler_tube
[taxprofiler_tube.pdf](https://github.com/nf-core/taxprofiler/files/12109121/taxprofiler_tube.pdf)
taxprofiler_tube

**Comment by LilyAnderssonLee** [comment]
@jfy133 Apparently I need to modify the regular expression used within the MetaPhlAn4 module in order to run the MetaPhlAn3 database since these two versions of the database include different file formats. 
Original code:
```
BT2_DB=`find -L "${metaphlan_db_latest}" -name "*rev.1.bt2l" -exec dirname {} \\;`
BT2_DB_INDEX=`find -L ${metaphlan_db_latest} -name "*.rev.1.bt2l" | sed 's/\\.rev.1.bt2l\$//' | sed 's/.*\\///'`
```
Modified version:
```
BT2_DB=`find -L "${metaphlan_db_latest}" -name "*rev.1.bt2*" -exec dirname {} \\;`
BT2_DB_INDEX=`find -L ${metaphlan_db_latest} -name "*.rev.1.bt2*" | sed 's/\\.rev.1.bt2.*\$//' | sed 's/.*\\///'`
```
I will create a PR to update the MetaPhlAn4 module if it's ok with you.

**Comment by jfy133** [comment]
Yup sure!

**Pull request merged**